src/core/query.ts

```diff
@@ -452,39 +452,47 @@ export class Query<TResult, TError> {
     const isFetchingMore = fetchMore ? (previous ? 'previous' : 'next') : false
     const prevPages: TResult[] = (this.state.data as any) || []
 
-    // Create function to fetch a page
-    const fetchPage = async (
+    // Create function to fetch pages
+    const fetchPages = async (
       pages: TResult[],
+      numberOfPages: number,
       prepend?: boolean,
-      cursor?: unknown
+      initialCursor?: unknown
     ) => {
-      const lastPage = getLastPage(pages, prepend)
+      let i = 0
+      let newPages = [...pages]
+      let lastPage = getLastPage(newPages, prepend)
+      let cursor = initialCursor
 
       if (
         typeof cursor === 'undefined' &&
         typeof lastPage !== 'undefined' &&
         config.getFetchMore
       ) {
-        cursor = config.getFetchMore(lastPage, pages)
+        cursor = config.getFetchMore(lastPage, newPages)
       }
 
-      const page = await config.queryFn(...params, cursor)
+      while (i < numberOfPages) {
+        lastPage = await config.queryFn(...params, cursor)
```

Comment:
This will still generate a lot of polyfill code because of the async await usage in iterations.

But I think the only change that is really needed in the original code is:
```js
cursor = config.getFetchMore(lastPage, pages)
```
to:
```js
cursor = config.getFetchMore(lastPage, pages)

if (typeof cursor === 'undefined' || cursor === false) {
  return pages
}
```

Reply:
That's a nice idea, I will update the code. 👍